### PR TITLE
Update type of ConstructorOptions to make sence

### PR DESCRIPTION
**Pull request by kuboon**
Sorry! I forgot to change the type definition!

### CODE REVIEW

**Comment by Errorname** [comment]
Nice catch! Thanks for the PR 🙂

File: src/types.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ export type ID = string
 export type AccessToken = string
 
 export interface ConstructorOptions {
-  token: () => AccessToken
+  token: () => Promise<AccessToken>
```
Comment (kuboon) [inline src/types.ts]:
```suggestion
  token: () => Promise<AccessToken> | AccessToken
```

Comment (kuboon) [reply src/types.ts]: should I do this for backward compatibility?

Comment (kuboon) [reply src/types.ts]: await simply returns non Promise value, but Typescript complains if token() returns non Promise value.

Comment (Errorname) [reply src/types.ts]: Yes, I'll add it for the release

Comment (Errorname) [reply src/types.ts]: I just released 1.2.1 that should fix this

**Pull request merged**